Lib/_pyrepl/base_eventqueue.py

```diff
@@ -73,14 +73,19 @@ def push(self, char: int | bytes | str) -> None:
         """
         Processes a character by updating the buffer and handling special key mappings.
         """
-        ord_char = char if isinstance(char, int) else ord(char)
-        if ord_char > 255:
-            assert isinstance(char, str)
-            char = bytes(char.encode(self.encoding, "replace"))
+
+        if isinstance(char, bytes):
             self.buf.extend(char)
         else:
-            char = bytes(bytearray((ord_char,)))
-            self.buf.append(ord_char)
+
+            ord_char = char if isinstance(char, int) else ord(char)
+            if ord_char > 255:
+                assert isinstance(char, str)
+                char = bytes(char.encode(self.encoding, "replace"))
+                self.buf.extend(char)
+            else:
+                char = bytes(bytearray((ord_char,)))
```

Comment:
I know this was just moved, but could we simplify it while we're here?
```suggestion
                char = bytes([ord_char])
```

Reply:
Sure, thanks!